crates/bvs-vault-router/src/contract.rs

```diff
@@ -325,13 +326,149 @@ mod execute {
+            registry.clone(),
+            &bvs_registry::msg::QueryMsg::SlashingParameters {
+                service: info.sender.to_string(),
+                timestamp: Some(slash_req.request.timestamp.seconds()),
+            },
+        )?;
+
+        let slashing_parameters = match slashing_parameters {
+            Some(p) => p,
+            None => {
+                return Err(InvalidSlashingRequest {
+                    msg: "Service has not enabled slashing at timestamp.".to_string(),
+                })
+            }
+        };
+
+        let accused_operator = deps
+            .api
+            .addr_validate(slash_req.request.operator.as_str())?;
+
+        // Require active status between operator and service
+        let StatusResponse(operator_service_status) = deps.querier.query_wasm_smart(
+            registry.to_string(),
+            &bvs_registry::msg::QueryMsg::Status {
+                service: info.sender.to_string(),
+                operator: accused_operator.to_string(),
+                timestamp: Some(slash_req.request.timestamp.seconds()),
+            },
+        )?;
+
+        // Prevent any active service from locking slash of slash request that it doesn't belong to
+        if operator_service_status != u8::from(RegistrationStatus::Active) {
+            return Err(InvalidSlashingRequest {
+                msg: "Service and Operator are not active at timestamp.".to_string(),
+            });
+        }
+
+        let now = env.block.time;
+        let cond_expired = now > slash_req.request_expiry;
+        let cond_not_aged = now
+            < slash_req
+                .request_time
+                .plus_seconds(slashing_parameters.resolution_window);
+
+        if cond_expired {
+            return Err(ContractError::InvalidSlashingRequest {
+                msg: "Slash id is expired".to_string(),
+            });
+        };
+
+        if cond_not_aged {
+            return Err(ContractError::InvalidSlashingRequest {
+                msg: "Resolution window for this id has not passed".to_string(),
+            });
+        };
+
+        let vaults_managed = state::OPERATOR_VAULTS
+            .prefix(&accused_operator)
+            .keys(deps.storage, None, None, cosmwasm_std::Order::Ascending)
+            .collect::<StdResult<Vec<_>>>()?;
+
+        // Array instead of map to avoid the overheads
+        let mut messages = Vec::new();
+        for vault in vaults_managed {
+            let vault_info = vault::get_vault_info(deps.as_ref(), &vault)?;
+            let total_assets = vault_info.total_assets;
+            let bips = Uint128::from(slash_req.request.bips as u128);
+
+            let slash_percent = bips.checked_div(Uint128::from(100_u128)).unwrap();
+
+            // multiply_ratio is always floored.
+            let slash_absolute =
+                total_assets.multiply_ratio(slash_percent, Uint128::from(100_u128));
```

Comment:
You're unnecessarily losing precision here.

```suggestion
            // Due to the nature of the integer division involved, the result is always floored.
            let slash_absolute = total_assets.multiply_ratio(bips, Uint128::from(10000_u128));
```

Reply:
Oh I see what you mean now. will fix.

Reply:
@kgmyatthu actually I don't see the fix in the code yet. have this change been comitted?

Reply:
The patch didn't got in, earlier. Fixed now.